src/shims/unix/linux/eventfd.rs

```diff
@@ -112,11 +115,13 @@ impl FileDescription for Event {
         self_ref: &FileDescriptionRef,
         _communicate_allowed: bool,
         bytes: &[u8],
+        dest: &MPlaceTy<'tcx>,
         ecx: &mut MiriInterpCx<'tcx>,
-    ) -> InterpResult<'tcx, io::Result<usize>> {
+    ) -> InterpResult<'tcx> {
         // Check the size of slice, and return error only if the size of the slice < 8.
         let Some(bytes) = bytes.first_chunk::<U64_ARRAY_SIZE>() else {
-            return Ok(Err(Error::from(ErrorKind::InvalidInput)));
+            let result = Err(Error::from(ErrorKind::InvalidInput));
+            return ecx.return_written_byte_count_or_error(result, dest);
         };
         // Convert from bytes to int according to host endianness.
         let num = match ecx.tcx.sess.target.endian {
```

Comment:
This is very strange... why do we now have completely different code paths for `read` and `write` here? That indicates an oddly asymmetric API: `read` gets a `Pointer`, but `write` not. Why that? Seems better to do this consistently. Here we'd really like to also just do `ptr_to_mplace_unaligned` and then read from that place.